# Análisis del comportamiento meteorológico y variables climáticas (2012)

## 2da Pre-Entrega

## GRUPO 9


*   Oscar Luis Gallardo
*   Gabriel Ollier
*   Augusto Rosso




## Objetivos
Explorar y comprender la distribución de las variables meteorológicas (temperatura, humedad, presión, radiación), identificar patrones estacionales a lo largo del año, detectar y limpiar anomalías en las mediciones, y encontrar correlaciones clave entre los distintos fenómenos climáticos.

- reconocer la estructura y el significado de un dataset;
- diferenciar variables numéricas, categóricas, temporales e identificadores;
- detectar faltantes, duplicados, formatos inconsistentes y valores atípicos;
- justificar decisiones de limpieza según el contexto;
- transformar variables y crear nuevas características interpretables.


## Preparación: cargar el dataset

In [ ]:
# impotamos la libreria y le asignamos el alias correspondiente.
import pandas as pd

df = pd.read_csv('/content/informacion-meteorologica-2012.csv')

# Parte 1 — Análisis de Datos

## Conociendo nuestro dataset



In [ ]:
display(df.shape)#dimensiones del dataset

(29559, 11)

In [ ]:
df.head(10)#vista inicial

,fecha,hora,estacion,vv_m_s,dv,temp_c,hr_porc,press_mbar,pluv_mm,rad_sol_w_m2,uv_uvindex
0,2012-01-01,00:00:00,1283,0.4,67.5,24.6,65,1010.8,0,0,0.0
1,2012-01-01,00:00:00,1284,2.7,135.0,24.8,60,1011.6,0,0,0.0
2,2012-01-01,00:00:00,1286,0.0,112.5,24.8,60,1012.6,0,0,0.0
3,2012-01-01,00:00:00,1291,0.0,180.0,24.6,59,1011.5,0,0,0.0
4,2012-01-01,00:00:00,1296,0.4,45.0,24.9,57,1010.4,0,0,0.0
5,2012-01-01,00:00:00,1305,0.4,90.0,25.1,61,1010.5,0,0,0.0
6,2012-01-01,00:00:00,1309,3.6,90.0,24.1,71,1011.7,0,0,0.0
7,2012-01-01,00:00:00,1310,2.7,90.0,24.7,60,1012.0,0,0,0.0
8,2012-01-01,00:00:00,1314,1.8,90.0,25.1,58,1010.7,0,0,0.0
9,2012-01-01,00:00:00,1318,0.4,67.5,1802.6,60,1011.9,0,0,0.0


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29559 entries, 0 to 29558
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   fecha         29559 non-null  object 
 1   hora          29559 non-null  object 
 2   estacion      29559 non-null  int64  
 3   vv_m_s        29559 non-null  float64
 4   dv            29559 non-null  float64
 5   temp_c        29559 non-null  float64
 6   hr_porc       29559 non-null  int64  
 7   press_mbar    29559 non-null  float64
 8   pluv_mm       29559 non-null  int64  
 9   rad_sol_w_m2  29559 non-null  int64  
 10  uv_uvindex    29559 non-null  float64
dtypes: float64(5), int64(4), object(2)
memory usage: 2.5+ MB


In [ ]:
display(df.dtypes)

,0
fecha,object
hora,object
estacion,int64
vv_m_s,float64
dv,float64
temp_c,float64
hr_porc,int64
press_mbar,float64
pluv_mm,int64
rad_sol_w_m2,int64


In [ ]:
df.isnull().sum()

,0
fecha,0
hora,0
estacion,0
vv_m_s,0
dv,0
temp_c,0
hr_porc,0
press_mbar,0
pluv_mm,0
rad_sol_w_m2,0


In [ ]:
df.duplicated().sum()

np.int64(0)

In [ ]:
df[df.duplicated()]

,fecha,hora,estacion,vv_m_s,dv,temp_c,hr_porc,press_mbar,pluv_mm,rad_sol_w_m2,uv_uvindex


# Conclusiones de la Exploración Inicial

- El dataset cuenta con 29,559 registros y 11 variables.
- No se detectaron valores nulos ni filas duplicadas, lo que permite avanzar sin necesidad de aplicar técnicas de imputación.
- Las variables numéricas están correctamente en formatos float64 e int64. Las columnas temporales (fecha y hora) están separadas y en formato de texto (object), lo que requiere una transformación a datetime para poder realizar análisis estacionales.
- Sin entrar mucho en detalle, simplemente imprimiendo unas filas se detectó un error físico en las mediciones. La fila de índice 9 registra un valor en temp_c de 1802.6 °C. Esto es algo que vamos a tener en cuenta en el analisís futuro.

##  Identificación de variables


In [ ]:
df.columns

Index(['fecha', 'hora', 'estacion', 'vv_m_s', 'dv', 'temp_c', 'hr_porc',
       'press_mbar', 'pluv_mm', 'rad_sol_w_m2', 'uv_uvindex'],
      dtype='object')

## Diccionario de variables

| Variable | Significado esperado | Tipo conceptual |
|---|---|---|
| `fecha` |Fecha en la que se realizó la medición | Temporal |
| `hora` |	Hora exacta de la medición |	Temporal
| `estación` |	Identificador de la estación meteorológica |	Identificador
| `vv_m_s` |	Velocidad del viento en metros por segundo |	Numérica continua
| `dv` |	Dirección del viento (en grados) |	Numérica continua
| `temp_c` |	Temperatura en grados Celsius |	Numérica continua
| `hr_proc` |	Humedad relativa en porcentaje |	Numérica discreta
| `press_mbar` |	Presión atmosférica en milibares |	Numérica continua
| `pluv_mm` |	Precipitación pluvial (lluvia) en milímetros |	Numérica discreta
| `rad_sol_w_m2` |	Radiación solar en vatios por metro cuadrado |	Numérica discreta
| `uv_uvindex` |	Índice ultravioleta (UV) |	Numérica continua


In [ ]:
df["estacion"].unique()

array([1283, 1284, 1286, 1291, 1296, 1305, 1309, 1310, 1314, 1318, 1316,
       1295, 1289, 1304, 1315])

In [ ]:
df["estacion"].value_counts()

,count
estacion,
1286,3385
1289,2713
1310,2571
1284,2548
1291,2313
1318,2115
1296,2030
1305,1984
1309,1847


Existe un desbalance en la cantidad de mediciones por estación. Las tres estaciones que más datos aportaron al dataset son la 1286 (3.385 registros), la 1289 (2.713 registros) y la 1310 (2.571 registros). Por el contrario, la estación 1295 es la que menos actividad registró, con apenas 850 mediciones.

# Parte 2 — Transformación y limpieza de datos

## Conversión de fechas

`fecha` y `hora` representan fechas, aunque inicialmente estén almacenadas como texto.



In [ ]:
df_copia = df.copy()

fecha_nueva = pd.to_datetime(df["fecha"], format="mixed")
hora_nueva = pd.to_datetime(df["hora"], format="mixed")
df_copia["fecha"] = fecha_nueva
df_copia["hora"] = hora_nueva

In [ ]:
df_copia.dtypes

,0
fecha,datetime64[ns]
hora,datetime64[ns]
estacion,int64
vv_m_s,float64
dv,float64
temp_c,float64
hr_porc,int64
press_mbar,float64
pluv_mm,int64
rad_sol_w_m2,int64


# Parte 3 -  Deteccion de outliers

Defino cuartiles y rango intercuartilico para ver en el dataframe los valores atipicos.

In [ ]:
Q1 = df['temp_c'].quantile(0.25)
Q3 = df['temp_c'].quantile(0.75)
IQR = Q3 - Q1
limite_inferior = Q1 - 1.5 * IQR
limite_superior = Q3 + 1.5 * IQR
print(f"Q1: {Q1}\nQ3: {Q3}\nIQR: {IQR}\nLimite inferior: {limite_inferior}\nLimite superior: {limite_superior}")

Q1: 16.1
Q3: 25.6
IQR: 9.5
Limite inferior: 1.8500000000000014
Limite superior: 39.85


In [ ]:
outliers_temp = df[(df['temp_c'] < limite_inferior) | (df['temp_c'] > limite_superior)]
display(outliers_temp)
outliers_temp.count()
outliers_temp.groupby('temp_c').count()

,fecha,hora,estacion,vv_m_s,dv,temp_c,hr_porc,press_mbar,pluv_mm,rad_sol_w_m2,uv_uvindex
9,2012-01-01,2026-09-17 00:00:00,1318,0.4,67.5,1802.6,60,1011.9,0,0,0.0
59,2012-01-01,2026-09-17 05:00:00,1318,0.4,90.0,1802.6,255,1012.8,0,0,0.0
99,2012-01-01,2026-09-17 09:00:00,1318,0.9,67.5,1802.6,68,1015.1,0,459,10.3
129,2012-01-01,2026-09-17 12:00:00,1318,1.3,90.0,1802.6,62,1014.2,0,933,15.8
139,2012-01-01,2026-09-17 13:00:00,1318,1.3,90.0,1802.6,62,1013.4,0,997,16.0
...,...,...,...,...,...,...,...,...,...,...,...
29112,2012-07-08,2026-09-17 05:00:00,1286,0.4,270.0,1802.6,55,1018.8,0,0,0.0
29222,2012-07-10,2026-09-17 12:00:00,1286,0.9,270.0,1802.6,65,1012.0,0,52,2.0
29234,2012-07-10,2026-09-17 18:00:00,1286,1.3,270.0,1802.6,60,1014.0,0,4,0.0
29290,2012-07-11,2026-09-17 22:00:00,1286,0.9,270.0,1802.6,54,1006.9,0,0,0.0


,fecha,hora,estacion,vv_m_s,dv,hr_porc,press_mbar,pluv_mm,rad_sol_w_m2,uv_uvindex
temp_c,,,,,,,,,,
1.1,2,2,2,2,2,2,2,2,2,2
1.3,4,4,4,4,4,4,4,4,4,4
1.4,2,2,2,2,2,2,2,2,2,2
1.5,1,1,1,1,1,1,1,1,1,1
1.6,1,1,1,1,1,1,1,1,1,1
1.7,4,4,4,4,4,4,4,4,4,4
1802.6,600,600,600,600,600,600,600,600,600,600


El valor maximo de 1806.2 en temp_c se repite en 600 filas, lo cual es raro que se replique en un volumen así. Estos valores de primera son fisicamente imposibles en un contexto normal, por lo que podria ser un error de lectura, de decimales, o escala.

Este valor atipíco afecta la media y la desviacion estandar. La desviacion estandar tiene un valor de 251.39, a pesar de que el 75% de los valores estan por debajo de 25.6, esto arrastra la media al valor de 56.72, frente a una mediana de 21.3..

# 4. Variable objetivo

> **La variable objetivo es la temperatura ambiental (`temp_c`), y el objetivo del proyecto es pronosticarla y estimarla con precisión analizando su comportamiento en función de factores meteorológicos clave, como la presión atmosférica, la radiación solar y la humedad.**